// Copyright 2019 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

package shaka.proto;

option optimize_for = LITE_RUNTIME;

// Describes info about a test media file.
message MediaInfo {
  // Describes the expected fields of an EncodedFrame.
  message Frame {
    required double pts = 1;
    required double dts = 2;
    optional double duration = 3;
    optional bool is_encrypted = 4;

    // The hexadecimal hash of the "data" field.
    required string data_hash = 5;
  }

  // Describes a stream that the frames belong to.
  message Stream {
    // The numerator from the time_scale.
    required uint32 time_scale_num = 1;
    // The denominator from the time_scale.
    required uint32 time_scale_den = 2;
    // Whether the stream is video.
    required bool is_video = 3;

    // The hexadecimal hash of the extra_data field.
    required string extra_data_hash = 4;

    // The numerator from the sample aspect ratio.
    required uint32 sar_num = 5;
    // The denominator from the sample aspect ratio.
    required uint32 sar_den = 6;
  }

  // The MIME type for the content.
  required string mime = 1;
  // The frames that appear in this file.
  repeated Frame frames = 2;
  // The stream all the frames belong to.
  required Stream stream = 3;
}
